diff --git a/target.cpp b/target.cpp
index 82bf65c..501a2ef 100644
--- a/target.cpp
+++ b/target.cpp
@@ -348,7 +348,7 @@ bool Target::findClangIntrinsicHeaders(std::string &path) {
             auto &file = dir;
 
             intrindir = dir.str();
-            file << "/xmmintrin.h";
+            file << "/include/xmmintrin.h";
 
             if (fileExists(file.str())) {
               if (cv > *clangversion) {
@@ -361,13 +361,9 @@ bool Target::findClangIntrinsicHeaders(std::string &path) {
             return false;
           };
 
-          tmp << dir.str() << "/" << file << "/include";
+          tmp << dir.str() << "/" << file;
 
-          if (!checkDir(tmp)) {
-            clear(tmp);
-            tmp << dir.str() << "/" << file;
-            checkDir(tmp);
-          }
+          checkDir(tmp);
         }
         return true;
       }
@@ -389,6 +385,11 @@ do {                                                                           \
 #define TRYDIR2(libdir) TRYDIR(clangbindir, libdir)
 #define TRYDIR3(libdir) TRYDIR(std::string(), libdir)
 
+  char *env = getenv("OSXCROSS_CLANG_INTRINSIC_PATH");
+  if (env != NULL) {
+    TRYDIR3(env);
+  }
+
 #ifdef __CYGWIN__
 #ifdef __x86_64__
   TRYDIR2("/../lib/clang/x86_64-pc-cygwin");
@@ -722,7 +722,7 @@ bool Target::setup() {
         return false;
       }
 
-      fargs.push_back("-isystem");
+      fargs.push_back("-resource-dir");
       fargs.push_back(tmp);
     }
