diff --git a/target.cpp b/target.cpp
index 82bf65c..cb48bd7 100644
--- a/target.cpp
+++ b/target.cpp
@@ -699,6 +699,14 @@ bool Target::setup() {
 
   fargs.push_back(compilerexecname);
 
+  char *linkerpath = getenv("OSXCROSS_LINKERPATH");
+  if (linkerpath != NULL)
+    fargs.push_back(std::string("--ld-path=") + linkerpath);
+
+  char *linkertype = getenv("OSXCROSS_LINKERTYPE");
+  if (linkertype != NULL)
+    fargs.push_back(std::string("-fuse-ld=") + linkertype);
+
   if (isClang()) {
     std::string tmp;
