diff --git a/target.cpp b/target.cpp
index 82bf65c..9d2df60 100644
--- a/target.cpp
+++ b/target.cpp
@@ -40,9 +41,14 @@
 
 namespace target {
 
+const char *getTarget() {
+    const char *env = getenv("OSXCROSS_TARGET");
+    return env ? env : getDefaultTarget();
+}
+
 Target::Target()
     : vendor(getDefaultVendor()), SDK(getenv("OSXCROSS_SDKROOT")),
-      arch(Arch::x86_64), target(getDefaultTarget()), stdlib(StdLib::unset),
+      arch(Arch::x86_64), target(getTarget()), stdlib(StdLib::unset),
       usegcclibs(), wliblto(-1), compiler(getDefaultCompilerIdentifier()),
       compilername(getDefaultCompilerName()), language() {
   if (!getExecutablePath(execpath, sizeof(execpath)))
